### ISSUE TITLE
Allow specify cpu or memory resource separately

### ISSUE THREAD

**Issue by hzxuzhonghu** [istio/istio#12249]
**Describe the bug**
When I create a deployment with sidecar auto inject enabled. The sidecar container can not be injected.

```
apiVersion: extensions/v1beta1
kind: Deployment
metadata:
  name: sleep
spec:
  replicas: 1
  template:
    metadata:
      labels:
        app: sleep
    annotations:
        sidecar.istio.io/proxyCPU: "200m"
    spec:
      containers:
      - name: sleep
        image: pstauffer/curl
        command: ["/bin/sleep", "3650d"]
        imagePullPolicy: IfNotPresent
```
With the error `Error creating: Internal error occurred: admission webhook "sidecar-injector.istio.io" denied the request: error unmarshaling JSON: quantities must match the regular expression '^([+-]?[0-9.]+)([eEinumkKMGTP]*[-+]?[0-9]*)$'`

But when I specify both `sidecar.istio.io/proxyCPU` and `sidecar.istio.io/proxyMemory`. It can be successful.

**Comment by howardjohn** [comment]
https://github.com/istio/istio/blob/de06caf5c2c20aae7c20476026906a5618940362/install/kubernetes/helm/istio/templates/sidecar-injector-configmap.yaml#L225-L229
If you have just proxyCPU it sets both CPU and memory

**Comment by howardjohn** [comment]
Fixed in #12250
